packages/broker/src/StorageNodeRegistry.ts

```diff
@@ -24,14 +31,14 @@ export class StorageNodeRegistry {
         return new StorageNodeRegistry(urlByAddress, config.streamrUrl)
     }
 
-    async getUrlByStreamId(streamId: string): Promise<string> {
-        const storageNodeAddress = await this.getStorageNodeAddress(streamId)
-        if (storageNodeAddress !== undefined) {
-            const url = this.getUrlByAddress(storageNodeAddress)
-            if (url !== undefined) {
-                return url
+    async getUrlsByStreamId(streamId: string): Promise<string[]> {
+        const storageNodeAddresses = await this.getStorageNodeAddress(streamId)
+        if (storageNodeAddresses !== undefined) {
+            const urls = this.getUrlsByAddresses(storageNodeAddresses)
+            if (urls.length > 0) {
+                return urls
```

Comment:
Should we print the warning to the logs if any of the addresses is not in the registry? 

If we'd iterate over each address (instead of urls) in `historicalData.createResponse`, the fetching of the URL for an address could be part of an iteration loop. -> We'd try another address if we can't get an URL for an address, or that URL responses non-200.

Reply:
Would be worth a warning yeah if an address is not in the registry.